torch_xla/core/xla_model.py

```diff
@@ -446,8 +447,14 @@ def all_reduce(reduce_type, inputs, scale=1.0, groups=None, pin_layout=True):
   """
   groups = groups or []
   if isinstance(inputs, torch.Tensor):
-    result = torch_xla._XLAC._xla_all_reduce(reduce_type, inputs, scale, groups,
-                                             pin_layout)
+    result = None
+    if scale == 1.0 and groups == [] and pin_layout:
+      # TODO(alanwaketan): Support groups.
+      result = torch.ops.c10d_functional.all_reduce(inputs, reduce_type, "", [],
```

Comment:
ranks=[] and world_size=0?  seems like you are not quite using the API as intended, just curious why

Reply:
Oh, the intention of this PR is to integrate the case where only one group in the whole world. Will follow up with the correct implementation of adopting the ranks and world_size later. In PyTorch/XLA, our runtime manages those information for us and not the users. Therefore, we usually don't need to pass down any ranks/world_size information down except for grouping.

Reply:
Can we add a comment here explain that only c10d op is traceable for the dynamo?

Reply:
Sure, will do.